docs/sql-ref-functions-udf-aggregate.md

```diff
@@ -94,8 +94,45 @@ For example, a user-defined average for untyped DataFrames can look like:
 <div data-lang="java"  markdown="1">
   {% include_example untyped_custom_aggregation java/org/apache/spark/examples/sql/JavaUserDefinedUntypedAggregation.java%}
 </div>
+<div data-lang="sql"  markdown="1">
+{% highlight sql %}
```

Comment:
Quick question, what about simply we add such examples in `UserDefinedUntypedAggregation` and `JavaUserDefinedUntypedAggregation` like https://github.com/apache/spark/pull/28087/files/8736f5cc8eb19f13b19d710571e3f203f0c7f292#diff-eb8e5f2dc83e098ff1d8068a69946389R43-R44?

Reply:
Do you mean add sql examples for the following scala UDF too?
```
    spark.udf().register("random", random);
    spark.sql("SELECT random()").show();
```

Reply:
I actually meant to just add the example:

```diff
--- a/examples/src/main/scala/org/apache/spark/examples/sql/UserDefinedTypedAggregation.scala
+++ b/examples/src/main/scala/org/apache/spark/examples/sql/UserDefinedTypedAggregation.scala
@@ -18,7 +18,8 @@ package org.apache.spark.examples.sql

 // $example on:typed_custom_aggregation$
 import org.apache.spark.sql.{Encoder, Encoders, SparkSession}
 import org.apache.spark.sql.expressions.Aggregator
+import org.apache.spark.sql.functions.udaf
 // $example off:typed_custom_aggregation$

 object UserDefinedTypedAggregation {
@@ -81,6 +82,14 @@ object UserDefinedTypedAggregation {
     // +--------------+
     // |        3750.0|
     // +--------------+
+
+    spark.udf.register("average_salary", udaf(MyAverage))
+    ds.selectExpr("average_salary()").show()
+    // +--------------+
+    // |average_salary|
+    // +--------------+
+    // |        3750.0|
+    // +--------------+
     // $example off:typed_custom_aggregation$
```

in `UserDefinedUntypedAggregation` and `JavaUserDefinedUntypedAggregation` .. to look similar with the examples in `JavaUserDefinedScalar.java` and `UserDefinedScalar.scala`

Reply:
Thanks for the clarification. I will fix this.

Reply:
Ah .. damn it. I am sorry. I didn't notice there are two separate classes `UserDefinedTypedAggregation` and UserDefinedUntypedAggregation`.  There's the example I wanted already at `UserDefinedUntypedAggregation`